packages/core/src/selectedGroups.spec.ts

```diff
@@ -0,0 +1,47 @@
+import { TestEnvironment } from "./tests/index";
+
+// import the settings for the css prefixes
+import settings from "carbon-components/es/globals/js/settings";
+
+import { options } from "./configuration";
+import { Events } from "./interfaces";
+
+import { select } from "d3-selection";
+
+describe("selectedGroups option", () => {
+	beforeEach(function () {
+		const testEnvironment = new TestEnvironment();
+		testEnvironment.render();
+
+		this.chart = testEnvironment.getChartReference();
+		this.testEnvironment = testEnvironment;
+	});
+
+	describe("set default selectedGroups", () => {
+		it("should match the array length in options", function (done) {
+			const numberOfSelectedGroup = this.chart.model.getOptions().data.selectedGroups.length;
```

Comment:
there's no point in comparing with a `selectedGroups` array that's gone into the library. You'd need to compare it to what you actually provide in options, because that way we can actually detect if at some point we incorrectly modify `selectedGroups` in model

Reply:
I suggest storing the array you include in your options in a variable

Reply:
Ok, got it.